src/pybind/mgr/orchestrator.py

```diff
@@ -1026,12 +1025,29 @@ class PlacementSpec(object):
     """
     For APIs that need to specify a node subset
     """
-    def __init__(self, label=None, nodes=None):
+    def __init__(self, label=None, nodes=None, count=None):
         self.label = label
         if nodes:
-            self.nodes = [parse_host_specs(x, require_network=False) for x in nodes]
+            self.nodes = [parse_host_specs(x, require_network=False) for x in nodes if x]
         else:
             self.nodes = []
+        self.count = count if count else 1
```

Comment:
We should not invent any defaults here. 

```suggestion
        self.count = count  # type: Optional[int]
```